frontend/packages/console-app/src/redux/actions/__tests__/cloud-shell-actions.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { Dispatch } from 'react-redux';
+import {
+  setCloudShellExpanded,
+  setCloudShellActive,
+  Actions,
+  toggleCloudShellExpanded,
+} from '../cloud-shell-actions';
+import cloudShellConfirmationModal from '../../../components/cloud-shell/cloudShellConfirmationModal';
+
+const cloudShellConfirmationModalMock = cloudShellConfirmationModal as jest.Mock;
+
+jest.mock('../../../components/cloud-shell/cloudShellConfirmationModal', () => ({
+  default: jest.fn((action) => action()),
+}));
+
+describe('Cloud shell actions', () => {
+  beforeEach(() => {
+    cloudShellConfirmationModalMock.mockClear();
+  });
+
+  it('should create expand action', () => {
+    expect(setCloudShellExpanded(false)).toEqual({
+      type: Actions.SetCloudShellExpanded,
+      payload: {
+        isExpanded: false,
+      },
+    });
+    expect(setCloudShellExpanded(true)).toEqual({
+      type: Actions.SetCloudShellExpanded,
+      payload: {
+        isExpanded: true,
+      },
+    });
+  });
+
+  it('should create active action', () => {
+    expect(setCloudShellActive(false)).toEqual({
+      type: Actions.SetCloudShellActive,
+      payload: {
+        isActive: false,
+      },
+    });
+    expect(setCloudShellActive(true)).toEqual({
+      type: Actions.SetCloudShellActive,
+      payload: {
+        isActive: true,
+      },
+    });
+  });
+
+  it('should thunk dispatch toggle expand true action', () => {
+    const dispatch = jest.fn<Dispatch>();
+    // initial isExpanded state is false
+    const state = { plugins: { console: { cloudShell: { isExpanded: false } } } } as any;
+    toggleCloudShellExpanded()(dispatch, () => state);
+    expect(dispatch).toHaveBeenCalledWith({
+      type: Actions.SetCloudShellExpanded,
+      payload: {
+        isExpanded: true,
+      },
+    });
```

Comment:
To my previous comment, I think we can improve this so it doesn't test `action` and avoids the "extra key" manual test update.

I created this helper method to simplify the usage because Jest doesn't have "partial nested" matches in function returns.
```
const payloadContaining = (v: Object) => expect.objectContaining({ payload: expect.objectContaining(v) });
```
```suggestion
    expect(dispatch).toHaveBeenCalledWith(payloadContaining({ isExpanded: true }));
```